src/DynamoCore/UpdateManager/UpdateManager.cs

```diff
@@ -654,20 +654,34 @@ public void QuitAndInstallUpdate()
 
         public void HostApplicationBeginQuit()
         {
-            if (!string.IsNullOrEmpty(UpdateFileLocation))
+            if (string.IsNullOrEmpty(UpdateFileLocation)) return;
```

Comment:
In what cases would this be null or empty?  You might want a short clarifying comment because it's not apparent to me.